src/scenes/Activities/Favorite/store.js

```diff
@@ -0,0 +1,81 @@
+import { observable, action, runInAction, reaction } from 'mobx';
+import { uniqBy, difference, orderBy } from 'lodash';
+import { Routes, SortBy, OracleStatus } from 'constants';
+import { queryAllTopics, queryAllOracles } from '../../../network/graphql/queries';
+import Topic from '../../../stores/models/Topic';
+import Oracle from '../../../stores/models/Oracle';
+
+const INIT_VALUES_FAVPAGE = {
+  displayList: [], // data list
+  loading: true,
+};
+
+export default class FavoriteStore {
+  @observable favList = JSON.parse(localStorage.getItem('bodhi_dapp_favList')) || []; // Data example: '21e389b909c7ab977088c8d43802d459b0eb521a'
+
+  @observable displayList = INIT_VALUES_FAVPAGE.displayList
+  @observable loading = INIT_VALUES_FAVPAGE.loading
+
+  constructor(app) {
+    this.app = app;
+    reaction(
+      () => this.app.wallet.addresses + this.app.global.syncBlockNum,
+      () => {
+        if (this.app.ui.location === Routes.FAVORITE) {
+          this.init();
+        }
+      }
+    );
+    reaction(
+      () => this.favList,
+      async () => {
+        this.displayList = await this.fetchFav();
+      }
+    );
+  }
+
+  updateLocalStorageFavList() {
+    localStorage.setItem('bodhi_dapp_favList', JSON.stringify(this.favList));
+  }
+
+  @action
+  setFavorite = (topicAddress) => {
+    if (this.isInFavorite(topicAddress)) {
+      this.favList = this.favList.filter(x => x !== topicAddress);
+    } else {
+      this.favList.push(topicAddress);
+    }
+    this.updateLocalStorageFavList();
+  }
+
+  @action
+  isInFavorite = (topicAddress) => this.favList.find(x => x === topicAddress)
```

Comment:
you will have to use the `not not operator` to turn it to a boolean: 
`!!this.favList.find(x => x === topicAddress)`

believe this currently is returning the object itself, but looks like you want a boolean.